cheroot/ssl/builtin.py

```diff
@@ -277,6 +277,11 @@ def wrap(self, sock):
             s = self.context.wrap_socket(
                 sock, do_handshake_on_connect=True, server_side=True,
             )
+        except ssl.SSLZeroReturnError:
+            # This is almost certainly due to the cherrypy engine
+            # 'pinging' the socket to assert it's connectable;
+            # the 'ping' isn't SSL.
```

Comment:
You said that you've disabled the Checker and it's still happening. Pretty sure it's unrelated, then.
It must be this https://github.com/cherrypy/cheroot/blob/3f9b1cb/cheroot/ssl/builtin.py#L55-L83.
By the way, maybe the `suppress()` in that helper function needs to be extended too.

Reply:
It isn't the checker, but cherrypy has a call to a module called portend which is used to move the lifecycle of the service to the active state.  This is a simple responds to TCP port check.  So it is a check but not the checker...

Reply:
Oh, that's good to know. That would be useful to mention.